tutorials/basic-topics/sorting/bubble-sort.md

```diff
@@ -1,14 +1,149 @@
 ---
 title: 'Bubble Sort'
-description: ''
+description: 'It is sorting algorithm based on the idea of repeatedly comparing pairs of adjacent elements and then swapping their positions if they exist in the wrong order.'
 hide_table_of_contents: true
-draft: true
 keywords:
-  - leetcode
-  - tutorial
-  - sorting
-  - algorithm
-  - bubble sort
+- leetcode
+- tutorial
+- sorting
+- algorithm
+- bubble sort
 ---
 
-<TutorialAuthors names="@TBC"/>
+<TutorialAuthors  names="@RadhikaChhabra17"/>
+
+## Overview
+
+Bubble Sort is a sorting algorithm which compares the adjacent elements and swap their positions if they are placed in wrong order. At max, we need to compare adjacent elements for $$(n - 1)$$ iterations where $n$ is the size of array to be sorted. At the end of each iteration, larger (or smaller, as required) value is sorted and placed at correct positions.
+
+The syntax that we are using is `cpp`, please refer to your own language of preference if needed.
+
+Algorithm :
+1. Make two nested loop, the 1st loop would be for number of pass the algorithm woud run, total $n-1$ passes and second would be for comparison in that pass. In each pass, one element is sorted (largest or smallest, as required) and placed in correct position and rest are compared in further passes.
+2. If one element is bigger (or smaller in decreasing order case) than its next element, then both should be swapped. 
+
+Consider the example of unsorted list and see how the algorithm works.
+
+arr = {3,6,4,2,5}: 
+
+1st pass =>
+    
+	index = 0 -> 3,6,4,2,5 
+	index = 1 -> 3,4,6,2,5
+	index = 2 -> 3,4,2,6,5
+    index = 3 -> 3,4,2,5,6
+
+2nd pass => 
+
+	index = 0 -> 3,4,2,5,6
+	index = 1 -> 3,2,4,5,6
+	index = 2 -> 3,2,4,5,6
+
+3rd pass => 
+
+	index = 0 -> 2,3,4,5,6
+	index = 1 -> 2,3,4,5,6
+
+4th pass =>
+
+	index = 0 -> 2,3,4,5,6  
+	
+We can use a variable *check* to see if there is swap in one pass or not. If there is no swapping in one pass, they we don't have to check for other pass.
+
+ <Tabs>
+<TabItem value="c++" label="C++">
+<SolutionAuthor name="@RadhikaChhabra17"/>
+
+```c++
+void bubblesort(vector<int> &arr) {
+	int n = arr.size();
+	bool check = true;
+	for(int i = 0; i < n - 1 && check; i++) {
```

Comment:
there should be a space after `for`. please check all.

`for (int i = 0; i < n - 1 && check; i++) {`